### PR TITLE
Resolve `TokenUpdateSpecs` errors

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
This PR fixes all remaining tests in the TokenUpdateSpec, but two, as the last two require more time for debugging.
Changes include:
 - Correct response code, according to the tests for renewal period
 - Changes in the order of check, to be compliant with the tests in the spec
 - Changes to token create expiry setting
 - Add treasury as owner for specific nft, if owner is null

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by netopyr — COMMENTED** [review]
In general, this PR looks good. But it changes validation methods that are used in several places. Did you double-check that we have no regressions?

**Comment by georgi-l95** [comment]
@netopyr Actually fixes couple of tests in the TokenUpdateSpec, but breaks only 1 from TopicUpdateSpec (`validateMultipleFields()`) . So I'm not sure which is the correct logic, they both require different ResponseCodes for the same operation. Would like someone to confirm what is the correct logic.
Changes here fix more than 4-5 tests, but break 1.

**Comment by netopyr** [comment]
I wouldn't be surprised if both behaviors are correct. 🙂 The returned error codes are not always consistent in our API.

As a quick fix, we can add a second parameter to `validateAutoRenewPeriod()`, which allows us to specify the error code.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java [hunk 1]
```diff
@@ -240,9 +240,7 @@ private ExpiryMeta getExpiryMeta(
             @NonNull final HandleContext context) {
         final var maxEntityLifetime =
                 context.configuration().getConfigData(EntitiesConfig.class).maxLifetime();
-        final var impliedExpiry = op.hasAutoRenewPeriod()
-                ? consensusTime + op.autoRenewPeriod().seconds()
-                : consensusTime + maxEntityLifetime;
+        final var impliedExpiry = op.hasExpiry() ? op.expiry().seconds() : consensusTime + maxEntityLifetime;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:243]: Awesome. 👍

**Review by Neeharika-Sompalli — COMMENTED** [review]
As @netopyr mentioned, both are correct. Since we have to keep mono-service behavior for differential testing, we can make `TopicUpdateHandler`  return `AUTORENEW_DURATION_NOT_IN_RANGE` when it fails. and add a TODO comment to change in future.

**Review by Neeharika-Sompalli — COMMENTED** [review]
Need to merge back develop and see none of HAPI tests are failing.

**Comment by netopyr** [comment]
Unfortunately, there is a conflict now.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/handle/validation/AttributeValidatorImplTest.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ void rejectsBriefAutoRenewPeriod() {
 
         assertThatThrownBy(() -> subject.validateAutoRenewPeriod(55L))
                 .isInstanceOf(HandleException.class)
-                .has(responseCode(AUTORENEW_DURATION_NOT_IN_RANGE));
+                .has(responseCode(INVALID_RENEWAL_PERIOD));
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/handle/validation/AttributeValidatorImplTest.java:120]: Why are these tests being changed?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java [hunk 3]
```diff
@@ -228,7 +230,17 @@ private ExpiryMeta resolvedUpdateMetaFrom(
...
         if (updatesExpiryMeta(op)) {
             final var updateMeta = new ExpiryMeta(effExpiryOf(op), effAutoRenewPeriodOf(op), op.autoRenewAccount());
-            return expiryValidator.resolveUpdateAttempt(currentMeta, updateMeta);
+            try {
+                return expiryValidator.resolveUpdateAttempt(currentMeta, updateMeta);
+            } catch (final HandleException e) {
+                if (e.getStatus() == INVALID_RENEWAL_PERIOD) {
+                    // Tokens throw INVALID_EXPIRATION_TIME, but for topic it's expected currently to throw
+                    // AUTORENEW_DURATION_NOT_IN_RANGE
+                    // TODO: We need to change this in the future.
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:239]: `future('<github issue number>')` is preferable to `TODO`

Comment (georgi-l95) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java]: done.

**Review by mhess-swl — DISMISSED** [review]
LGTM!

**Review by netopyr — DISMISSED** [review]
LGTM

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetNftInfoHandler.java [hunk 4]
```diff
@@ -123,19 +125,22 @@ public Response findResponse(@NonNull final QueryContext context, @NonNull final
     private Optional<TokenNftInfo> infoForNft(
             @NonNull final NftID nftId,
             @NonNull final ReadableNftStore readableNftStore,
+            @NonNull final ReadableTokenStore readableTokenStore,
```
Comment (rbair23) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetNftInfoHandler.java:128]: Can you also update the javadoc comment with the missing param?

**Review by rbair23 — DISMISSED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetNftInfoHandler.java (hunk 4 above)
Comment (georgi-l95) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetNftInfoHandler.java:130]: Ah yes, forgot about javadoc, thank you. Updated.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**